packages/wonder-blocks-modal/components/modal-launcher/index.test.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+import React from "react";
+import {shallow} from "enzyme";
+
+import ModalLauncher from "./index.js";
+
+describe("ModalLauncher", () => {
+    test("Children can launch the modal", () => {
+        const wrapper = shallow(
+            <ModalLauncher modal={<div />}>
+                {({openModal}) => <button onClick={openModal} />}
+            </ModalLauncher>,
+        );
+        expect(wrapper.find("ModalLauncherPortal")).toHaveLength(0);
+        wrapper.find("button").simulate("click");
+        expect(wrapper.find("ModalLauncherPortal")).toHaveLength(1);
+    });
+
+    test("Modal can close itself after launching", (done) => {
+        let opened = false;
+
+        // Once the modal mounts, we'll immediately self-close it on the next
+        // tick, to test the children's ability to self-close.
+        //
+        // NOTE(mdr): It would be nice to have this be, like, a close button
+        //     that closes when you click it. But that requires the button to
+        //     actually _mount_, which means we'd need to do a full DOM render
+        //     including `ModalLauncherPortal`, and that seems to be going
+        //     beyond the scope of this test. Really we just want to check that
+        //     this function receives a `closeModal` argument that works.
```

Comment:
`enzyme` has an option to do a full render if we wanted to.  I think the test you have are sufficient though.

Reply:
Yeah, my concern here isn't really to do with feasibility so much as keeping the test well-scoped. It would be odd for a bug in `ModalLauncherPortal` to trigger a failure in the "modal contents can self-close" test. It might be appropriate to have an integration test that does a full render like that, but clearly labeled as an integration test?